obspy/io/nied/knet/core.py

```diff
@@ -0,0 +1,312 @@
+    :type buf: Open file or open file like object.
+    """
+    first_string = buf.read(11).decode()
+    # File has less than 11 characters
+    if len(first_string) != 11:
+        return False
+    if first_string == 'Origin Time':
+        return True
+    return False
+
+
+def _prep_hdr_line(name, line):
+    """
+    Helper function to check the contents of a header line and split it.
+
+    :param name: String that the line should start with.
+    :type name: str
+    :param line: Line to check and split.
+    :type line: str
+    """
+    if not line.startswith(name):
+        raise KnetFormatError("Expected line to start with %s but got %s "
+                              % (name, line))
+    else:
+        return line.split()
+
+
+def _read_knet_hdr(hdrlines, convert_stnm=False, **kwargs):
+    """
+    Read the header values into a dictionary.
+
+    :param hdrlines: List of the header lines of a a K-NET/KiK-net ASCII file
+    :type hdrlines: list
+    :param convert_stnm: For station names with 6 letters write the last two
+    letters of the station code to the 'location' field
+    :type convert_stnm: bool
+    """
+    hdrdict = {'knet': {}}
+    hdrnames = ['Origin Time', 'Lat.', 'Long.', 'Depth. (km)', 'Mag.',
+                'Station Code', 'Station Lat.', 'Station Long.',
+                'Station Height(m)', 'Record Time', 'Sampling Freq(Hz)',
+                'Duration Time(s)', 'Dir.', 'Scale Factor', 'Max. Acc. (gal)',
+                'Last Correction', 'Memo.']
+    _i = 0
+    # Event information
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dt = flds[2] + ' ' + flds[3]
+    dt = UTCDateTime(datetime.strptime(dt, '%Y/%m/%d %H:%M:%S'))
+    # All times are in Japanese standard time which is 9 hours ahead of GMT
+    dt -= 9 * 3600.
+    hdrdict['knet']['evot'] = dt
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    lat = float(flds[1])
+    hdrdict['knet']['evla'] = lat
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    lon = float(flds[1])
+    hdrdict['knet']['evlo'] = lon
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dp = float(flds[2])
+    hdrdict['knet']['evdp'] = dp
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    mag = float(flds[1])
+    hdrdict['knet']['mag'] = mag
+
+    # Station information
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    # K-NET and KiK-Net station names can be more than 5 characters long
+    # which will cause the station name to be truncated when writing the
+    # the trace as miniSEED; if convert_stnm is enabled, the last two
+    # letters of the station code are written to the 'location' field
+    stnm = flds[2]
+    location = ''
+    if convert_stnm and len(stnm) > 5:
+        location = stnm[-2:]
+        stnm = stnm[:-2]
+    if len(stnm) > 7:
+        raise KnetFormatError("Station name can't be more than 7 characters\
+        long!")
```

Comment:
I prefer just starting the string indented on the next line, and I think it should all fit.